string addType
AISShip ship
int32 waypointIndex
